salt/modules/cmdmod.py

```diff
@@ -562,15 +562,22 @@ def _run(
 
             msg = "env command: {}".format(env_cmd)
             log.debug(log_callback(msg))
-            env_bytes, env_encoded_err = subprocess.Popen(
-                env_cmd,
-                stderr=subprocess.PIPE,
-                stdout=subprocess.PIPE,
-                stdin=subprocess.PIPE,
-            ).communicate(salt.utils.stringutils.to_bytes(py_code))
+            try:
+                env_bytes, env_encoded_err = subprocess.Popen(
+                    env_cmd,
+                    stderr=subprocess.PIPE,
+                    stdout=subprocess.PIPE,
+                    stdin=subprocess.PIPE,
+                ).communicate(salt.utils.stringutils.to_bytes(py_code), timeout=10)
+            except subprocess.TimeoutExpired:
```

Comment:
Can want to warn the user something when wrong?

Reply:
The `marker_count = 0` in the except clause will trigger the `log.error` statement on L583 so I don't think anything special needs to be added?